### PR TITLE
NFS for FSAL_RGW

### PR DESCRIPTION
**Pull request by dang**
Add support for FSAL_RGW to the NFS gateway.  Both standard and
containerized versions are supported.

### CODE REVIEW

File: roles/ceph-nfs/tasks/docker/copy_configs.yml [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+    dest: "{{ item.0 }}"
+    owner: "64045"
+    group: "64045"
+    mode: 0644
+  changed_when: false
   with_together:
     - ceph_config_keys
     - statconfig.results
-  when: item.1.stat.exists == false
+  when: item.1.stat.exists == true
```
Comment (leseb) [inline roles/ceph-nfs/tasks/docker/copy_configs.yml]: why changing this to `false`?

File: roles/ceph-nfs/tasks/docker/create_configs.yml [hunk 2]
```diff
@@ -7,6 +7,21 @@
     group: root
     mode: 0644
 
+- name: Create the NFS RGW user
```
Comment (leseb) [inline roles/ceph-nfs/tasks/docker/create_configs.yml]: for consistency purpose, please remove all the capital letters

File: roles/ceph-nfs/tasks/docker/create_configs.yml [hunk 3]
```diff
@@ -7,6 +7,21 @@
...
+    image: "{{ ceph_rgw_docker_username }}/{{ ceph_rgw_docker_imagename }}:{{ ceph_rgw_docker_image_tag }}"
+    name: ceph-{{ ansible_hostname }}-rgw-user
+    expose: "{{ ceph_rgw_civetweb_port }}"
+    ports: "{{ ceph_rgw_civetweb_port }}:{{ ceph_rgw_civetweb_port }}"
+    state: running
+    env: "CEPH_DAEMON=RGW_USER,RGW_USERNAME={{ ceph_nfs_rgw_user }},RGW_USER_ACCESS_KEY={{ ceph_nfs_rgw_access_key }},RGW_USER_SECRET_KEY={{ ceph_nfs_rgw_secret_key }}"
+    volumes: "/var/lib/ceph:/var/lib/ceph,/etc/ceph:/etc/ceph"
+  when: fsal_rgw
+
+- name: Get user create output
```
Comment (leseb) [inline roles/ceph-nfs/tasks/docker/create_configs.yml]: ditto

File: roles/ceph-nfs/tasks/pre_requisite.yml [hunk 4]
```diff
@@ -10,6 +10,32 @@
     - /var/lib/nfs/ganesha
     - /var/run/ganesha
 
+- name: create RGW NFS user
+  command: radosgw-admin user create --uid={{ ceph_nfs_rgw_user }} --display-name="RGW NFS User"
+  register: rgwuser
+  when: fsal_rgw
+
+- name: Set access key
```
Comment (leseb) [inline roles/ceph-nfs/tasks/pre_requisite.yml]: ditto

File: roles/ceph-nfs/tasks/pre_requisite.yml [hunk 5]
```diff
@@ -10,6 +10,32 @@
...
+  set_fact:
+    ceph_nfs_rgw_access_key: "{{ (rgwuser.stdout | from_json)['keys'][0]['access_key'] }}"
+  when: fsal_rgw
+
+- name: Set secret key
```
Comment (leseb) [inline roles/ceph-nfs/tasks/pre_requisite.yml]: ditto

**Comment by leseb** [comment]
Minor comments, generally looks good.

File: roles/ceph-nfs/tasks/docker/copy_configs.yml (hunk 1 above)
Comment (dang) [inline roles/ceph-nfs/tasks/docker/copy_configs.yml]: We need to copy all the files that aren't there yet.  copy_configs.yaml wasn't included before (it was a straight copy of the RGW file), so the fact that it was wrong before didn't matter.

**Pull request merged**